### ISSUE TITLE
Makefile:151: recipe for target 'all' failed [EosioClang-stamp/EosioClang-build]

### ISSUE THREAD

**Issue by eosauthority** [EOSIO/eosio.wasmsdk#14]
On Ubuntu 18.04 with new build and latest version of EOS which was running fine

which clang-4.0 | xargs ls -lah
lrwxrwxrwx 1 root root 25 Feb 13 08:26 /usr/bin/clang-4.0 -> ../lib/llvm-4.0/bin/clang

which clang++-4.0 | xargs ls -lah
lrwxrwxrwx 1 root root 27 Feb 13 08:26 /usr/bin/clang++-4.0 -> ../lib/llvm-4.0/bin/clang++

llvm-config-4.0 --version --has-rtti
4.0.1
YES

Steps executed before the error

> git clone --recursive https://github.com/eosio/eosio.wasmsdk
> cd eosio.wasmsdk
> ./build.sh EOS

Please see last few lines before the error

> [ 80%] Building CXX object tools/clang/lib/StaticAnalyzer/Checkers/CMakeFiles/clangStaticAnalyzerCheckers.dir/UndefinedArraySubscriptChecker.cpp.o
> [ 80%] Building CXX object tools/clang/lib/StaticAnalyzer/Checkers/CMakeFiles/clangStaticAnalyzerCheckers.dir/UndefinedAssignmentChecker.cpp.o
> [ 80%] Building CXX object tools/clang/lib/StaticAnalyzer/Checkers/CMakeFiles/clangStaticAnalyzerCheckers.dir/UnixAPIChecker.cpp.o
> [ 80%] Building CXX object tools/clang/lib/StaticAnalyzer/Checkers/CMakeFiles/clangStaticAnalyzerCheckers.dir/UnreachableCodeChecker.cpp.o
> [ 80%] Building CXX object tools/clang/lib/StaticAnalyzer/Checkers/CMakeFiles/clangStaticAnalyzerCheckers.dir/VforkChecker.cpp.o
> [ 80%] Building CXX object tools/clang/lib/StaticAnalyzer/Checkers/CMakeFiles/clangStaticAnalyzerCheckers.dir/VLASizeChecker.cpp.o
> [ 80%] Building CXX object tools/clang/lib/StaticAnalyzer/Checkers/CMakeFiles/clangStaticAnalyzerCheckers.dir/ValistChecker.cpp.o
> [ 80%] Building CXX object tools/clang/lib/StaticAnalyzer/Checkers/CMakeFiles/clangStaticAnalyzerCheckers.dir/VirtualCallChecker.cpp.o
> [ 80%] Linking CXX static library ../../../../../lib/libclangStaticAnalyzerCheckers.a
> [ 80%] Built target clangStaticAnalyzerCheckers
> Makefile:151: recipe for target 'all' failed
> make[3]: *** [all] Error 2
> CMakeFiles/EosioClang.dir/build.make:111: recipe for target 'EosioClang-prefix/src/EosioClang-stamp/EosioClang-build' failed
> make[2]: *** [EosioClang-prefix/src/EosioClang-stamp/EosioClang-build] Error 2
> CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/EosioClang.dir/all' failed
> make[1]: *** [CMakeFiles/EosioClang.dir/all] Error 2
> Makefile:129: recipe for target 'all' failed
> make: *** [all] Error 2

==========

Will try Ubuntu 16.04 also

**Comment by liamcurry** [comment]
I can't compile on Ubuntu 18.04 or 16.04. Here is the latest error I'm seeing on 16.04:

![image](https://user-images.githubusercontent.com/77758/43163511-59141502-8f5c-11e8-9006-403fc55567dd.png)

**Comment by datudou** [comment]
I think the memory is the key issue. When I first tried to compile it  in the vm which memory was 14GB. Got the same error as yours. 
After that  I set 64GB memory  for my vm instance( Ubuntu 16.04) . And then it could be compiled successfully.

**Comment by eosauthority** [comment]
> After that I set 64GB memory for my vm instance( Ubuntu 16.04) . And then it could be compiled successfully.

Same for me. Tried a 128GB memory virtual instance and the build on 18.04 worked as expected.

**Comment by liamcurry** [comment]
> I think the memory is the key issue.

Yes I think you're right. That error I got was on a Docker container with 8GB of RAM. Does anyone know the minimum amount of RAM required? Seems like it's more than 14GB based on @datudou's comments.

**Comment by rohanagarwal94** [comment]
@liamcurry did you manage to resolve this? Just a heads up, I only have 8 Gb RAM.

**Comment by rohanagarwal94** [comment]
@liamcurry thanks for the help.
